src/resolve/resolved-dns-dnssec.c

```diff
@@ -1242,8 +1242,10 @@ int dnssec_nsec3_hash(DnsResourceRecord *nsec3, const char *name, void *ret) {
                 return r;
 
         err = gcry_md_open(&md, algorithm, 0);
-        if (gcry_err_code(err) != GPG_ERR_NO_ERROR || !md)
-                return -EIO;
+        if (gcry_err_code(err) != GPG_ERR_NO_ERROR || !md) {
+                r = -EIO;
+                goto finish;
+        }
```

Comment:
hmm, are you saying gcry_md_open() initializes md even on failure? what a mess...

Reply:
I don't know - but Coverity seems to think so :-)